sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -423,7 +423,14 @@ case class InsertIntoHiveTable(
     // Attempt to delete the staging directory and the inclusive files. If failed, the files are
     // expected to be dropped at the normal termination of VM since deleteOnExit is used.
     try {
-      createdTempDir.foreach { path => path.getFileSystem(hadoopConf).delete(path, true) }
+      createdTempDir.foreach { path =>
+        path.getFileSystem(hadoopConf).delete(path, true)
+        // If we successfully delete the staging directory, remove it from FileSystem's cache.
+        val fs = path.getFileSystem(hadoopConf)
```

Comment:
You call getFileSystem twice; you can reuse it.
I suppose this adds an additional check to see if the file exists, but this won't be called that often.